### ISSUE TITLE
Getting covidcg working on third party computer

### ISSUE THREAD

**Issue by philt31** [vector-engineering/covidcg#244]
Hello,
this was an attempt to get covidcg working on my laptop. Version cloned is commit 591eed4aacd29aeec59fdc09dfc457ff42cef042.
To get snakemake to complete I made the following changes as shown below as difference between master and my branch:

```
diff --git a/cg_scripts/process_ack.py b/cg_scripts/process_ack.py
index 50fed2b..fca85c6 100644
--- a/cg_scripts/process_ack.py
+++ b/cg_scripts/process_ack.py
@@ -52,7 +52,7 @@ def process_ack(ack_files):
         ack_map["Authors"]
         .str.extract(r"^(?:[^,，;\s]+\s)*([^,，;]+)[,，;]")
         .squeeze()  # DataFrame of one column -> Series
-        .str.strip()
+        .strip()
     )
     # If the format is Last-Name Initials, ...
     # i.e., FirstName L-N, or FirstName L.N, or FirstName L.N., or FirstName LN
@@ -70,7 +70,7 @@ def process_ack(ack_files):
             r"^([\w’\'\-]+\s?[\w’\'\-]*)[,;\s](?:[^,，;]+)?[A-Z]{1}[-\.]{0,2}[A-Z]?\.?[,，;]"
         )
         .squeeze()
-        .str.strip()
+        .strip()
     )
```

I've just changed .str.strip() to .strip(). This may have to do with the acknowledgement file I was using not having the correct data.

```
diff --git a/cg_scripts/process_patient_metadata.py b/cg_scripts/process_patient_metadata.py
index c57e36f..4dd9372 100644
--- a/cg_scripts/process_patient_metadata.py
+++ b/cg_scripts/process_patient_metadata.py
@@ -456,7 +456,10 @@ def clean_specimen_metadata(patient_meta_df):
     print("Cleaning specimen metadata...", end="", flush=True)

     # Basic cleanup
-    patient_meta_df["specimen"] = patient_meta_df["Specimen"].str.strip()
+    # in test set data is nan so cast as str
+    patient_meta_df["specimen"] = patient_meta_df["Specimen"]
+    if not patient_meta_df["Specimen"].isnull().all():
+        patient_meta_df["specimen"] = patient_meta_df["Specimen"].str.strip()

     specimen_key_map = {
         "Alveolar lavage fluid": [],
@@ -721,6 +724,7 @@ def clean_specimen_metadata(patient_meta_df):
     patient_meta_df["specimen"] = patient_meta_df["specimen"].map(specimen_map)

     print("done")
+    '''
     print(
         'Setting "Specimen" values to "Unknown": {}'.format(
             ", ".join(
@@ -736,6 +740,7 @@ def clean_specimen_metadata(patient_meta_df):
             )
         )
     )
+    '''

     patient_meta_df["specimen"] = patient_meta_df["specimen"].fillna("Unknown")
```

Again I had issues here with null data so only used str.strip() if there was data in column and then commented out print cmd.

```
diff --git a/cg_scripts/process_locations.py b/cg_scripts/process_locations.py
index 707fc99..040c61c 100644
--- a/cg_scripts/process_locations.py
+++ b/cg_scripts/process_locations.py
@@ -4110,13 +4110,15 @@ def clean_location_data(location_df):
                 vals = [vals]

             # Turn each value into a logical mask
-            vals = [location_df[key] == v for v in vals]
+            #vals = [location_df[key] == v for v in vals]
+            vals = [False for v in vals]
             # Combine logical masks with logical ORs, and merge into the master mask with AND
             loc_mask = loc_mask & reduce(lambda x, y: (x | y), vals)

         # Set the output rules on the matching entries from loc_mask
         for out_key in output_rule.keys():
             location_df.loc[loc_mask, out_key] = output_rule[out_key]
+            # location_df.loc[loc_mask, out_key] = location_df[key]

     # Done
     return location_df    

```

I hacked this just to get the code compiling but I think it may have lead to an issue in the browser. The browser loads but i get the following js error:

```
location.js?c960:35 Uncaught TypeError: Cannot read property 'location_id' of undefined
    at getLocationIdsFromNode (location.js?c960:35)
    at eval (location.js?c960:48)
    at Array.forEach (<anonymous>)
    at getLocationIds (location.js?c960:47)
    at processCaseData (caseData.worker.js?5b91:238)
    at eval (caseData.worker.js?5b91:873)

```

**Comment by philt31** [comment]
Below are the sequences used in this test:
>hCoV-19/USA/MI-MDHHS-SC21953/2020|EPI_ISL_566004|2020-03-31
>hCoV-19/USA/MI-MDHHS-SC21954/2020|EPI_ISL_566005|2020-08-25
>hCoV-19/USA/MI-MDHHS-SC21955/2020|EPI_ISL_566006|2020-08-12
>hCoV-19/USA/MI-MDHHS-SC21956/2020|EPI_ISL_566007|2020-03-14
>hCoV-19/USA/MI-MDHHS-SC21957/2020|EPI_ISL_566008|2020-08-16
>hCoV-19/USA/MI-MDHHS-SC21958/2020|EPI_ISL_566009|2020-08-25
>hCoV-19/USA/MI-MDHHS-SC21959/2020|EPI_ISL_566010|2020-04-16
>hCoV-19/USA/MI-MDHHS-SC21960/2020|EPI_ISL_566011|2020-03-31
>hCoV-19/USA/MI-MDHHS-SC21961/2020|EPI_ISL_566012|2020-03-20
>hCoV-19/USA/MI-MDHHS-SC21962/2020|EPI_ISL_566013|2020-06-23
>hCoV-19/USA/MI-MDHHS-SC21963/2020|EPI_ISL_566014|2020-08-12
>hCoV-19/USA/MI-MDHHS-SC21964/2020|EPI_ISL_566015|2020-04-08
>hCoV-19/USA/MI-MDHHS-SC21965/2020|EPI_ISL_566016|2020-08-18
>hCoV-19/USA/MI-MDHHS-SC21966/2020|EPI_ISL_566017|2020-08-20

**Comment by atc3** [comment]
OK, I see the issue here with some of the metadata processing. With so few sequences, my CSV loader (pandas) is not casting some columns to strings as all the values are missing. I'll fix these with explicit casts

I suspect the error with the `location_id` is also missing data related - as the sequences you've selected are missing the fourth level of detail that is hard-coded into a lot of the scripts (i.e., Region --> Country --> Division --> Location. The sequences given are specified to Michigan, the "Division", but have no "Location).

I will try rebuilding from the sequences listed to reproduce/confirm the error

**Comment by atc3** [comment]
Sorry, going to keep this open until you confirm your issues have been resolved.

I downloaded the sequences you listed and reproduced the errors, and have the app working on my end with that data.

I've implemented some of your pipeline fixes in this pull request: [https://github.com/vector-engineering/covidcg/pull/245](https://github.com/vector-engineering/covidcg/pull/245)

The javascript error you were experiencing (where the app didn't load) was because the app tries to select sequences in USA + Canada when the app loads. Since you didn't have any sequences from Canada, that caused the app to crash. This has been fixed and (hopefully) requires no extra changes on your end.
